exercicios/para-casa/entregas/ewelyn-samaris/models/SavingAccount.model.ts

```diff
@@ -0,0 +1,50 @@
+import BankAccount from './BankAccount.model';
+import Customer from './Customer.model';
+import { AccountType } from './enums/AccountType.enum';
+import { TransactionStatus } from './enums/TransactionStatus.enum';
+import { TransactionType } from './enums/TransactionsType.enum';
+import Transaction from './Transaction.model';
+
+export default class SavingAccount extends BankAccount{
+    private MINIMUN_TO_DEPOSIT: number = 1;
+
+    constructor(owner: Customer, initialBalance: number) {
+        super(owner, AccountType.SAVING_ACCOUNT);
+        if(initialBalance) this.deposit(initialBalance);
+    }
+
+    deposit(value: number): void {

```

Comment:
I think this point can be improved and kept the code cleaner, here is a suggestion: 
deposit(value: number): void {
    const status = value < this.MINIMUN_TO_DEPOSIT ? TransactionStatus.ERROR : TransactionStatus.SUCCESS;
    const transaction = new Transaction(TransactionType.DEPOSIT, status, value, this.balance);

    if (status === TransactionStatus.SUCCESS) {
        this.accountBalance += value;
    }

    this.transactionHistory.push(transaction);
}
If it makes sense to you